### PR TITLE
Implement "Introduce named variable for captured block" rendering strategy

### PR DESCRIPTION
**Pull request by msalzburg**
This PR implements the "Introduce named variable for captured block" strategy from Issue #32. 

It defines a #block and a #block? method to access the captured, nested block content.

### CODE REVIEW

**Comment by jensljungblad** [comment]
@msalzburg Looking at https://github.com/github/actionview-component, it seems like they went with this strategy as well, but named the variable `content`. Should we do the same perhaps?

**Comment by jensljungblad** [comment]
We could also add a test case for validating its presence.

**Comment by msalzburg** [comment]
Hi @jensljungblad - I updated this branch by merging your small refactoring  and renamed the block handling methods to .content and .content? (again). Since the content block is optional I am not sure which presence you would like to validate? 

Maybe we can squeeze this feature in your upcoming 1.0.0 release?

File: README.md [hunk 1]
```diff
@@ -269,23 +269,27 @@ end
...
-  <div class="card__footer">
-    <%= card.footer %>
-  </div>
 </div>
 ```
 
 Elements can be thought of as isolated subcomponents, and they are defined on the component. Passing `multiple: true` makes it a repeating element, and passing a block lets us declare attributes on our elements, in the same way we declare attributes on components.
 
-In order to populate them with data, we pass a block to the component helper, which yields the component, which lets us set attributes and blocks on the element in the same way we do for components:
+In order to populate them with data, we pass a content block to the component helper, which yields the component, which lets us set attributes and content blocks on the element in the same way we do for components:
```
Comment (jensljungblad) [inline README.md]:
```suggestion
In order to populate them with data, we pass a block to the component helper, which yields the component, which lets us set attributes and content blocks on the element in the same way we do for components:
```

File: README.md [hunk 2]
```diff
@@ -122,13 +122,13 @@ end
...
 
-To inject some text or HTML content into our component we can print the component variable in our template, and populate it by passing a block to the component helper:
+To inject some text or HTML content into our component we can print the component .content method in our template, and populate it by passing a content block to the component helper:
 
 ```erb
 <% # app/components/alert/_alert.html.erb %>
 
 <div class="alert alert--<%= alert.context %>" role="alert">
-  <%= alert %>
+  <%= alert.content %>
```
Comment (jensljungblad) [inline README.md]:
```suggestion
  <%= alert.content %>
```

**Comment by jensljungblad** [comment]
I'll give you a chance to reply to my comment above but I'm happy to merge as is :)

**Comment by jensljungblad** [comment]
Eh, decided to just merge it :) Maybe I'll have a go at adding the tests.

**Comment by jensljungblad** [comment]
Thanks a lot for the contribution!

**Comment by jensljungblad** [comment]
@msalzburg Released as `1.0.0.rc2`.

**Comment by msalzburg** [comment]
Awesome - thanks!

**Pull request merged**